### DISCUSSION TITLE
How to set text color to be transparent? (Negative potato print)

### DISCUSSION THREAD

**Discussion by TS60** [typst/typst#4514] [Q&A] [answered]
Is it possible with PDF to set the "text color" of a given text to be 100 % transparent?
For a given text, only the area around the text should get printed and _not_ the text itself.

```typst
#rect(fill: luma(200))[
  This text should be in "transparent color", it should _not_ get printed,
  only the background of this rectangle should get printed in the specified color.
]
```

Note: this is _not_ the same as setting the text color to white. What happens when the paper color is not white?

![transparent](https://github.com/typst/typst/assets/152509161/4cee86e8-7df9-4b67-a111-f2242c9cad13)

Edit: To make it more clear: this is a normal typesetting technique often used in the past but nowadays used only rarely because most software has not implemented it. Here is an example: the paper shines through and the text is _not_ printed:

![Warmbrunn](https://github.com/typst/typst/assets/152509161/522e8d75-7e2c-4062-b1dd-dc2c0027000f)

![Thalbahn](https://github.com/typst/typst/assets/152509161/16f94bdd-eff3-4700-bf14-a47044f6d914)

I just found the [equivalent](https://tex.stackexchange.com/questions/618868/how-to-clip-text-from-a-drawn-rectangle-in-tikz) LaTeX question.

It seems [this](https://tex.stackexchange.com/a/188065) answer even uses PDF code.

**Comment by EpicEricEE** [comment]
For transparent but selectable text, you can use 
```typ
#rect(fill: luma(200))[
  #set text(fill: rgb(0, 0, 0, 0)) // Last number is opacity

  This text should be in "transparent color", it should _not_ get printed,
  only the background of this rectangle should get printed in the specified color.
]
```

If the text should also be unselectable, you can use the [`hide`](https://typst.app/docs/reference/layout/hide/) function.

**Reply by TS60** [reply]
Thank you for the answer. But unfortunately this will just hide the text. Instead it should be cut out of the rectangle so the paper color shines through after printed.

It seems this is called "Knockout" and is supported by PDF and SVG.

I updated the question and added example images I found.

**Comment by sitandr** [comment, marked as answer]
Seems like this should be impossible in Typst with currently available features. However, for the most use-cases it would be enough to set the text color to *page's background*. With context system it becomes possible, so the code would look like (probably `highlight` would fit better, but that depends on the required result):

```typ
#context highlight(text(stroke: page.fill)[Text])
```

If you have an image on background (or just put the text into a block with background), however, that wouldn't work and getting color at certain point would be hard. Hopefully, there would be ways to deal with that in future.

If this is a highly requested feature, maybe it is worth to create an issue to add it straight to the compiler.

**Comment by AhmedSalih3d** [comment]
This seems to be possible now in Typst v0.12!

`rgb(50,100,200).transparentize(50%)`

And of course set it to `0%` to get all the way away.

**Reply by ghost** [reply]
I don't think so. Do you have an example code?

**Comment by TS60** [comment]
An Issue is now created [here](https://github.com/typst/typst/issues/5403).
